### ISSUE TITLE
Specify namespace for events stored in etcd

### ISSUE THREAD

**Issue by saad-ali** [kubernetes/kubernetes#4079]
All kubelet events are currently stored in etcd with the default namespace. Consider changing the namespace for events to something more useful (pod namespace?).

Sample current etcd events:

```
root@kubernetes-master:/home/saadali# etcdctl -C 10.240.196.75:4001 ls --recursive /registry/events/
/registry/events/default
/registry/events/default/kibana-logging-controller-fplln.13bf8469c2af2212
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf8469cd8ac98d
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf846da8b2678c
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf846dbb77be5e
/registry/events/default/kibana-logging-controller-fplln.13bf846e005c5e5c
/registry/events/default/skydns-0133o.13bf847b0b9ef1b4
/registry/events/default/monitoring-heapster-controller-oh43e.13bf8466fc295dec
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf846700f196ab
/registry/events/default/skydns-0133o.13bf847d10234e89
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf848c1a989e7a
/registry/events/default/monitoring-heapster-controller-oh43e.13bf846da050226c
/registry/events/default/monitoring-heapster-controller-oh43e.13bf846e008afef6
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf847f35c101dd
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf848c2c6139e8
/registry/events/default/elasticsearch-logging-controller-gziey.13bf8494bb8984d9
/registry/events/default/monitoring-heapster-controller-oh43e.13bf8469c7f424bb
/registry/events/default/elasticsearch-logging-controller-gziey.13bf846d9cb83af8
/registry/events/default/monitoring-heapster-controller-oh43e.13bf8467eadb6d17
/registry/events/default/skydns-0133o.13bf84711087ec08
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf847f37530bf6
/registry/events/default/kibana-logging-controller-fplln.13bf8466433b7dc6
/registry/events/default/elasticsearch-logging-controller-gziey.13bf846681e8b2be
/registry/events/default/kibana-logging-controller-fplln.13bf846d9fe74bf0
/registry/events/default/monitoring-heapster-controller-oh43e.13bf847ee3c780d3
/registry/events/default/monitoring-heapster-controller-oh43e.13bf847ee74823a4
/registry/events/default/monitoring-heapster-controller-oh43e.13bf8466487703cc
/registry/events/default/kibana-logging-controller-fplln.13bf84667f2e6ae0
/registry/events/default/skydns-0133o.13bf846da3d77446
/registry/events/default/elasticsearch-logging-controller-gziey.13bf8466f96fbc68
/registry/events/default/skydns-0133o.13bf847b148d3df8
/registry/events/default/kibana-logging-controller-fplln.13bf8466f6de12da
/registry/events/default/skydns-0133o.13bf8467ed460e7c
/registry/events/default/elasticsearch-logging-controller-gziey.13bf8469c57702ad
/registry/events/default/elasticsearch-logging-controller-gziey.13bf846d9cde78de
/registry/events/default/skydns-0133o.13bf846da7920087
/registry/events/default/elasticsearch-logging-controller-gziey.13bf846645e08321
/registry/events/default/monitoring-heapster-controller-oh43e.13bf846684ad03f7
/registry/events/default/kibana-logging-controller-fplln.13bf847a9313e945
/registry/events/default/skydns-0133o.13bf847d0cfd7e6b
/registry/events/default/skydns-0133o.13bf847d0e8cf1b0
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf8467f0239bc4
/registry/events/default/kubernetes-minion-2.c.saad-dev-vms.internal.13bf84681ea4fbdb
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf84668931582c
/registry/events/default/kibana-logging-controller-fplln.13bf846d996ed74f
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf847f2e6dae09
/registry/events/default/elasticsearch-logging-controller-gziey.13bf8494c22df331
/registry/events/default/skydns-0133o.13bf84664a5e178b
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf84664cd92de4
/registry/events/default/skydns-0133o.13bf8469cabb473b
/registry/events/default/monitoring-heapster-controller-oh43e.13bf846da0d385ae
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf846dc7ae94fa
/registry/events/default/elasticsearch-logging-controller-gziey.13bf846ddca2aa3b
/registry/events/default/skydns-0133o.13bf847114195093
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf848c2e2a8aeb
/registry/events/default/skydns-0133o.13bf8466866577c4
/registry/events/default/kubernetes-minion-3.c.saad-dev-vms.internal.13bf84682e833fd8
/registry/events/default/elasticsearch-logging-controller-gziey.13bf8467e83c5a1e
/registry/events/default/skydns-0133o.13bf846de4be6f8e
/registry/events/default/kibana-logging-controller-fplln.13bf847a975151be
/registry/events/default/kubernetes-minion-1.c.saad-dev-vms.internal.13bf846816d2c085
/registry/events/default/skydns-0133o.13bf847b170e8640
/registry/events/default/monitoring-heapster-controller-oh43e.13bf846db7bea702
/registry/events/default/elasticsearch-logging-controller-gziey.13bf846df27e1b9c
/registry/events/default/kibana-logging-controller-fplln.13bf847a7a39a1ef
/registry/events/default/monitoring-heapster-controller-oh43e.13bf847ee5efbaef
/registry/events/default/skydns-0133o.13bf846da4418644
/registry/events/default/kibana-logging-controller-fplln.13bf846dbaab36a4
/registry/events/default/kubernetes-minion-4.c.saad-dev-vms.internal.13bf846966acb998
/registry/events/default/monitoring-influx-grafana-controller-ls6k1.13bf846e137a3534
/registry/events/default/skydns-0133o.13bf8470f70430bd
/registry/events/default/elasticsearch-logging-controller-gziey.13bf8494c057bec4
/registry/events/default/skydns-0133o.13bf8466fdea7158
/registry/events/default/kibana-logging-controller-fplln.13bf8467e5914020
/registry/events/file-6bb810db-kubernetes-minion-1
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846dad9479da
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846817a57449
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf84685a26d86c
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846888a7d58a
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf84688739484a
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846cc20a6eba
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846d057518d8
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846d06370552
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846d9c550409
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846e159fb06c
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf84681971b0d2
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf8468845d1864
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846e1964b05f
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846cfd4c62b0
/registry/events/file-6bb810db-kubernetes-minion-1/cadvisor-agent.13bf846deea260fc
/registry/events/file-6bb810db-kubernetes-minion-2
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf84682159e0c7
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf84688cec5aa3
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846d0d7002c4
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf84681fc78d24
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846d0e25064e
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846da0b8c35c
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846e183afb90
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846d02b4324d
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846e1cc2c28a
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf84685d51f4fb
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf8468869c8fe8
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf84688b8be1dd
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846cc9393ed5
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846db6b34258
/registry/events/file-6bb810db-kubernetes-minion-2/cadvisor-agent.13bf846dfecbea19
/registry/events/file-6bb810db-kubernetes-minion-4
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf846e53af5ff2
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf84709959649b
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf84696ecebc59
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8469ce2e65ce
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8469c8360ec4
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8469ccafae74
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf846e4d2b273a
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf84696b809965
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8469b1618e1a
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8470d0a5d5ba
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf846e7e09fa0e
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8470a274c477
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf8470c61250a9
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf846e11382d88
/registry/events/file-6bb810db-kubernetes-minion-4/cadvisor-agent.13bf84706a72621d
/registry/events/file-8cd71177-kubernetes-minion-1
/registry/events/file-8cd71177-kubernetes-minion-1/fluentd-to-elasticsearch.13bf846cbfc885c7
/registry/events/file-8cd71177-kubernetes-minion-1/fluentd-to-elasticsearch.13bf846cfcb39971
/registry/events/file-8cd71177-kubernetes-minion-1/fluentd-to-elasticsearch.13bf846d00604ec3
/registry/events/file-8cd71177-kubernetes-minion-1/fluentd-to-elasticsearch.13bf847a10720685
/registry/events/file-8cd71177-kubernetes-minion-1/fluentd-to-elasticsearch.13bf847a129080e7
/registry/events/file-8cd71177-kubernetes-minion-1/fluentd-to-elasticsearch.13bf847a1562ca91
/registry/events/file-8cd71177-kubernetes-minion-2
/registry/events/file-8cd71177-kubernetes-minion-2/fluentd-to-elasticsearch.13bf846cc808289b
/registry/events/file-8cd71177-kubernetes-minion-2/fluentd-to-elasticsearch.13bf846d01f3f341
/registry/events/file-8cd71177-kubernetes-minion-2/fluentd-to-elasticsearch.13bf846d08310037
/registry/events/file-8cd71177-kubernetes-minion-2/fluentd-to-elasticsearch.13bf84760d62c81f
/registry/events/file-8cd71177-kubernetes-minion-2/fluentd-to-elasticsearch.13bf847632b410b6
/registry/events/file-8cd71177-kubernetes-minion-2/fluentd-to-elasticsearch.13bf84763562e5fc
/registry/events/file-8cd71177-kubernetes-minion-3
/registry/events/file-8cd71177-kubernetes-minion-3/fluentd-to-elasticsearch.13bf8477711161c8
/registry/events/file-8cd71177-kubernetes-minion-3/fluentd-to-elasticsearch.13bf84777d9d8676
/registry/events/file-8cd71177-kubernetes-minion-3/fluentd-to-elasticsearch.13bf84777f4164ff
/registry/events/file-8cd71177-kubernetes-minion-3/fluentd-to-elasticsearch.13bf846cd7529aa8
/registry/events/file-8cd71177-kubernetes-minion-3/fluentd-to-elasticsearch.13bf846cdc495306
/registry/events/file-8cd71177-kubernetes-minion-3/fluentd-to-elasticsearch.13bf846d20d96950
/registry/events/file-8cd71177-kubernetes-minion-4
/registry/events/file-8cd71177-kubernetes-minion-4/fluentd-to-elasticsearch.13bf846e510f22dc
/registry/events/file-8cd71177-kubernetes-minion-4/fluentd-to-elasticsearch.13bf846e824acdbf
/registry/events/file-8cd71177-kubernetes-minion-4/fluentd-to-elasticsearch.13bf847abd2db520
/registry/events/file-8cd71177-kubernetes-minion-4/fluentd-to-elasticsearch.13bf847ac10fcf47
/registry/events/file-8cd71177-kubernetes-minion-4/fluentd-to-elasticsearch.13bf847ac38a8879
/registry/events/file-8cd71177-kubernetes-minion-4/fluentd-to-elasticsearch.13bf846e0fb9bf72
/registry/events/file-6bb810db-kubernetes-minion-3
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846e0cf2732c
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf84687916d2ee
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf8468a574db84
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846cdd6fd54f
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846cf248804b
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846d25029609
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846d26bacf14
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846d9fc454c2
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf84682f79519b
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf8468a07d9fc7
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846e2b6a989b
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846830aa2722
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf8468a7369abf
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846db7d8e514
/registry/events/file-6bb810db-kubernetes-minion-3/cadvisor-agent.13bf846e2598cfbf
```

**Comment by thockin** [comment]
Good point - they should probably be namespaced as per the related object,
but I STR some places where the related object is bogus...

**Comment by smarterclayton** [comment]
Pretty sure we fixed this in 1.2.

**Comment by falenn** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  curtis.l.bates@gmail.com
    Domain imwiz.com has exceeded the max emails per hour (181/150 (120%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o10.sgmail.github.com ([167.89.101.201]:35389)
	by box969.bluehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.87)
	(envelope-from <bounces+848413-5c7e-dev=imwiz.com@sgmail.github.com>)
	id 1dGTQ6-000XIM-NU
	for dev@imwiz.com; Thu, 01 Jun 2017 10:58:01 -0600
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=6cM4NEb3rN2i6VdXvqSyiJ277Yg=; b=IY/gnA4U0na/WZNe
	GNpCT8QPjQ3I7DAhb68NkKluVaUzLum0CbNuVJJMc51kpQeOgg/4SDdJ+1lTYkil
	u9aTJv5tdeOt7nEghEiEzmS8ZWAs+OPcSJqVttUsfhG3j4kvwLj7CYj/bk/rry62
	3XiMmlnlJ6ydRh32BwDf9SCkV+0=
Received: by filter0621p1mdw1.sendgrid.net with SMTP id filter0621p1mdw1-28200-5930474F-26
        2017-06-01 16:56:47.250794774 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0002p1iad1.sendgrid.net (SG) with ESMTP id Dy9ogQbTQxWn6Bg-5ytbJA
	for <dev@imwiz.com>; Thu, 01 Jun 2017 16:56:47.183 +0000 (UTC)
Date: Thu, 01 Jun 2017 09:56:47 -0700
